### PR TITLE
[TestEngine: Fix] use contract class to test

### PR DESCRIPTION
**Pull request by Jim8y**
This pr focus on enable us to use the contract class name to load and compile the test.

### CODE REVIEW

**Comment by Jim8y** [comment]
@shargon

File: tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <ProjectReference Include="..\..\src\Neo.SmartContract.Framework\Neo.SmartContract.Framework.csproj">
       <Aliases>scfx</Aliases>
     </ProjectReference>
+    <ProjectReference Include="..\Neo.Compiler.CSharp.TestContracts\Neo.Compiler.CSharp.TestContracts.csproj" />
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:12]: Aren't we going to have some warnings when compiling the tests?

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:12]: we are not compiling them, i am not sure i understand you, we just run them in unit tests as file... we dont compile them directly

Comment (shargon) [reply tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:12]: It's compiled

```
8>------ Build started: Project: Neo, Configuration: Debug Any CPU ------
6>C:\Neo\neo-devpack-dotnet\tests\Neo.SmartContract.Framework.TestContracts\Contract_ExtraAttribute.cs(9,30,9,34): warning CS0028: 'Contract_ExtraAttribute.Main(string, object[])' has the wrong signature to be an entry point
6>C:\Neo\neo-devpack-dotnet\tests\Neo.SmartContract.Framework.TestContracts\Contract_ExecutionEngine.cs(24,20,24,43): warning CS0618: 'Runtime.ScriptContainer' is obsolete: 'Use System.Runtime.Transaction instead'
6>C:\Neo\neo-devpack-dotnet\tests\Neo.SmartContract.Framework.TestContracts\Contract_Blockchain.cs(37,24,37,28): warning CS8603: Possible null reference return.
6>C:\Neo\neo-devpack-dotnet\tests\Neo.SmartContract.Framework.TestContracts\Contract_Blockchain.cs(75,24,75,28): warning CS8603: Possible null reference return.
````

Comment (Jim8y) [reply tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:12]: Ooops, dependenency..... but how you want warnings to be

**Review by cschuchardt88 — APPROVED** [review]

**Comment by shargon** [comment]
Conflicts

**Comment by Jim8y** [comment]
> Conflicts

Addressed

**Comment by Jim8y** [comment]
@shargon please review

File: tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj (hunk 1 above)
Comment (cschuchardt88) [reply tests/Neo.Compiler.CSharp.UnitTests/Neo.Compiler.CSharp.UnitTests.csproj:12]: warning are warning. Just the user decide how to treat them.

**Comment by shargon** [comment]
I want to move all the tests to the new test engine, it will be cleaner and user friendly, we need to do easily unit tests, otherwise the projects don't test his own projects

**Comment by Jim8y** [comment]
> I want to move all the tests to the new test engine, it will be cleaner and user friendly, we need to do easily unit tests, otherwise the projects don't test his own projects

Thats funny, how about i tell you i have a much better idea and you stop working on your engine?

**Pull request closed without merging**